static/js/components/search/util.js

```diff
@@ -100,7 +100,15 @@ export const NestedAccessorMixin = BaseSearchkitAccessorClass =>
         // Add the same 'AND' filter to query.index.filtersMap, with the nested path (not the uuid) as the key
         filtersMap[nestedPath] = groupedNestedFilter
         // If it exists, delete the key for this specific filter (since all filters on this path are grouped together).
-        delete filtersMap[this.getFilterMapKey()]
+        const oldKey = this.getFilterMapKey()
+        const matcher = matchFieldName(oldKey)
+        for (const key of Object.keys(filtersMap)) {
+          if (matcher(key)) {
+            delete filtersMap[key]
+            break
+          }
+        }
```

Comment:
This might be a little cleaner, but not a huge improvement so you can take or leave this

```javascript
const oldKey = this.getFilterMapKey()
const matcher = matchFieldName(oldKey)
filtersMap = _.omitBy(
  filtersMap
  (value, key) => ( matcher(key) )
);
```

Reply:
Here's a ramda implementation. 'Cleaner' but still kinda verbose and not exactly intuitive:

```javascript
R.compose(
  R.fromPairs,
  R.reject(
    R.compose(
      matcher,
      R.view(R.lensIndex(0))
    )
  ),
  R.toPairs
)(filtersMap);
```

Reply:
Sure